### PR TITLE
[WIP] IPLD Git integration

### PR DESCRIPTION
**Pull request by magik6k**
TODO:

- [ ] Use ipld-git from gx
  - Depends on https://github.com/ipfs/go-ipld-git/pull/2, https://github.com/ipfs/go-ipld-git/pull/5
- [ ] Docs (where would they go?)

`test/sharness/t0054-dag-git-data/git.tar.gz` contains objects generated using make-test-repo.sh from https://github.com/ipfs/go-ipld-git/pull/2/files#diff-dbf76111333473e75d09d3498a488606

### CODE REVIEW

File: test/sharness/t0054-dag-git.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+test_description="Test dag git object handling"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+test_expect_success "prepare test data" '
+	tar xzf ../t0054-dag-git-data/git.tar.gz
```
Comment (Kubuxu) [inline test/sharness/t0054-dag-git.sh]: Can you add here the comment where the file comes from.

Comment (magik6k) [inline test/sharness/t0054-dag-git.sh]: Will do when ipfs/go-ipld-git#2 is merged

**Comment by whyrusleeping** [comment]
Now we need to start an earnest discussion on how to handle objects larger than 2MB

**Comment by whyrusleeping** [comment]
we also need to have a discussion about the sha1 collision. cc @Stebalien

**Comment by Kubuxu** [comment]
This will be superseded by git plugin PR.

**Comment by magik6k** [comment]
Closing as plugin version was merged

**Pull request closed without merging**